test/syscalls/linux/tuntap.cc

```diff
@@ -413,6 +415,50 @@ TEST_F(TuntapTest, SendUdpTriggersArpResolution) {
   }
 }
 
+TEST_F(TuntapTest, TUNNoPacketInfo) {
```

Comment:
This only really tests reading. The data written to the TUN device should be read and checked somewhere to make sure write works. `PingKernel` above does a form of this, and also avoids raw sockets and thus `CAP_NET_RAW`.

Reply:
Thanks!
I missed it somehow...